aws/rust-runtime/aws-types/src/os_shim_internal.rs

```diff
@@ -32,7 +32,8 @@ use std::sync::Arc;
 ///     map
 /// });
 /// ```
-pub struct Fs(fs::Inner);
+#[derive(Clone)]
+pub struct Fs(Arc<fs::Inner>);
```

Comment:
Should we have `Inner` implement `Clone` and wrap the hash map with `Arc` rather than wrapping `Inner`? That way the real implementation doesn't need an `Arc` at all.

Edit: Looks like you're moving env away from that approach below. I think I missing something. What's your reasoning for moving the `Arc` up a level?

Reply:
yeah I did it to simplify but that's a really good point. By having an exterior arc, we're _always_ going to need to allocate. Will refactor.